### ISSUE TITLE
helm 3.8 - does not support latest tag

### ISSUE THREAD

**Issue by mitzen** [helm/helm#10670]
hi all, i am using helm 3.8 and when i try to pull chart using the following command :- 

helm pull oci://myazure.io/helm/my-chart --version latest

it get the error "improper constraint: latest"

Any help/info would be great. Thank you 

With helm3.7.1, there's no issue.

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
Azure container registry

**Comment by yxxhero** [comment]
@mitzen  maybe because of Semantic Version policy.

**Comment by mitzen** [comment]
thanks yxxhero - it could be due to that. Not really sure. 
Figure i ask the folks here :)

**Comment by yxxhero** [comment]
@mitzen from the code. I am sure this becase of Semantic Version policy.

**Comment by mitzen** [comment]
thanks yxxhero. after looking at the code, its due to package dependency to semver. 
https://github.com/Masterminds/semver/blob/master/constraints.go 

does this means, latest tag is not something that will be supported?

**Comment by bkendzior** [comment]
so this means that charts pushed with older versions (not using semver) can no longer be pulled with helm 3.8? What's the reasoning behind this new strictness?

**Comment by yxxhero** [comment]
@bkendzior from the code. it will find latest tag. you can test when your container registry container has the latest tag.

**Comment by mitzen** [comment]
@yxxhero  - i am confused. if you said it will pick up latest version, then why am i getting improper constraint?

**Comment by yxxhero** [comment]
@mitzen are you sure the you has the lastest tag?

**Comment by bacongobbler** [comment]
It sounds like some people are confusing the difference between "the literal tag 'latest'" and "the most recent version, referred to as 'the latest tag'". Allow me to explain a few concepts here.

In Helm, chart package versions are expressed as x.y.z, where x is the major version, y is the minor version, and z is the patch version, following Semantic Versioning terminology. Charts MUST comply with the semantic version range policy. Version names such as "latest" or "development" do not match with this policy.

To grab the most recent version of a chart, you omit the `--version` flag and Helm will fetch the most recent version.

Where possible, use version ranges instead of pinning to an exact version. The suggested default is to use a patch-level version match:

```
helm install mychart --version ~1.2.3
```

This will match version 1.2.3 and any patches to that release. In other words, ~1.2.3 is equivalent to >= 1.2.3, < 1.3.0.

Implementing version range support was one of the last features we had to implement in the OCI experimental flag. That is why it allowed invalid tags, but that was never intended as a solution.

To answer your question:

> so this means that charts pushed with older versions (not using semver) can no longer be pulled with helm 3.8?

Yes. Charts that were built prior to the OCI experimental flag being lifted (and do not comply with [HIP 6](https://github.com/helm/community/blob/main/hips/hip-0006.md#3-chart-versions--oci-reference-tags)) cannot be pulled with Helm 3.8.

That is why we marked it as an experimental feature.
